src/accountsdb/db.zig

```diff
@@ -1848,6 +1869,95 @@ pub const AccountsDB = struct {
         }
     }
 
+    // writes one account to storage
+    pub fn putAccount(self: *AccountsDB, account: Account, pubkey: Pubkey, slot: Slot) !void {
```

Comment:
@yewman was thinking this should take an AccountSharedData. I think it's reasonable, I think this function would only be called from the runtime anyway. Any thoughts?

Reply:
That makes sense to me. I have no idea how a caller would be able to construct an instance of `Account`. It seems like it only makes sense when it comes from AccountsDB.

Actually I think `AccountSharedData` should probably be the standard representation of accounts, and `Account` is just a specific thing that's used for reading accounts out of AccountsDB. Likewise I would probably rename `AccountSharedData` to `Account` and move it to `core`, and rename to `Account` to something like `AccountReadHandle` and move it to `accountsdb`.

Reply:
Also minor nitpick but I would order the parameters differently to make it consistent with other kv stores, like hashmaps.

```suggestion
    pub fn putAccount(self: *AccountsDB, pubkey: Pubkey, slot: Slot, account: Account) !void {
```

Reply:
Gone for AccountSharedData + slot as 1st param:
3ea90f21c4fabb99a1320a6b1204fb44f1c9dee5

I think eventually we should be able to merge Account and AccountSharedData, but requires some effort

Reply:
Yeah separate for now is definitely the move.

I think I misread your previous comment, I changed it from account/pubkey/slot to account/slot/pubkey. 

Thinking it should now be slot/pubkey/account, changed in b4441d16cfd520123c27a22f836690f0fc162b98